File: go/ssh_client.go
package main

import (
	"bytes"
	"fmt"
	"golang.org/x/crypto/ssh"
	"time"
)

// SSHClient struct to hold SSH connection details
type SSHClient struct {
	Host     string
	Port     string
	Username string
	Password string
}

// RunCommand connects to the SSH server and runs the specified command
func (client *SSHClient) RunCommand(command string) (string, error) {
	config := &ssh.ClientConfig{
		User: client.Username,
		Auth: []ssh.AuthMethod{
			ssh.Password(client.Password),
		},
		HostKeyCallback: ssh.InsecureIgnoreHostKey(),
		Timeout:         5 * time.Second,
	}

	// Connect to the SSH server
	connection, err := ssh.Dial("tcp", fmt.Sprintf("%s:%s", client.Host, client.Port), config)
	if err != nil {
		return "", fmt.Errorf("failed to connect to SSH server: %w", err)
	}
	defer connection.Close()

	// Create a session
	session, err := connection.NewSession()
	if err != nil {
		return "", fmt.Errorf("failed to create SSH session: %w", err)
	}
	defer session.Close()

	// Run the command and capture the output
	var outputBuffer bytes.Buffer
	session.Stdout = &outputBuffer
	if err := session.Run(command); err != nil {
		return "", fmt.Errorf("failed to run command: %w", err)
	}

	return outputBuffer.String(), nil
}

File: python/utils/ssh_client.py
import paramiko

class SSHClient:
    def __init__(self, host, port, username, password):
        self.host = host
        self.port = port
        self.username = username
        self.password = password

    def run_command(self, command):
        try:
            # Create an SSH client
            client = paramiko.SSHClient()
            client.set_missing_host_key_policy(paramiko.AutoAddPolicy())  # Insecure, for testing only

            # Connect to the SSH server
            client.connect(
                hostname=self.host,
                port=self.port,
                username=self.username,
                password=self.password,
                timeout=5
            )

            # Execute the command
            stdin, stdout, stderr = client.exec_command(command)
            output = stdout.read().decode().strip()
            error = stderr.read().decode().strip()

            # Get the exit status
            exit_status = stdout.channel.recv_exit_status()

            # Close the connection
            client.close()

            if exit_status != 0:
                return output, f"Command failed with exit status {exit_status}: {error}"
            return output, None
        except Exception as e:
            return "", str(e)